src/image.rs

```diff
@@ -1,25 +1,26 @@
 use anyhow::{anyhow, Result};
-use docker_api::api::{ContainerCreateOpts, PullOpts, RegistryAuth, RmContainerOpts};
-use docker_api::Images;
-use futures_util::{StreamExt, TryStreamExt};
-use podman_api::api::Images as PodmanImages;
-use podman_api::opts::ContainerCreateOpts as PodmanContainerCreateOpts;
-use podman_api::opts::PullOpts as PodmanPullOpts;
-use podman_api::opts::RegistryAuth as PodmanRegistryAuth;
-use std::path::PathBuf;
-use tar::Archive;
+use async_trait::async_trait;
 
+use super::docker::DockerImage;
+use super::podman::PodmanImage;
 use super::runtime;
 
-pub struct Image {
-    pub image: String,
-    pub repo: String,
-    pub tag: String,
-    pub runtime: runtime::Runtime,
+#[async_trait]
+pub trait Image {
+    async fn pull(&self, username: String, password: String, force: bool) -> Result<()>;
+    async fn start(&self) -> Result<String>;
+    async fn stop(&self, id: String) -> Result<()>;
+    async fn copy_files(
```

Comment:
nit: can we add some docs here outlining the trait and the various methods? This is part of the public API. 

As I mentioned, we may want to rename this module, or this trait, to make it more coherent. In reality, we are manipulating a container here, not an image. maybe something like `runtime::Container` as the module and trait name respectively makes more sense?